x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/termsenum/action/TransportTermsEnumAction.java

```diff
@@ -445,36 +499,41 @@ protected AsyncBroadcastAction(Task task, TermsEnumRequest request, ActionListen
             nodes = clusterState.nodes();
             logger.trace("resolving shards based on cluster state version [{}]", clusterState.version());
             nodeBundles = getNodeBundles(clusterState, request, concreteIndices);
-            expectedOps = nodeBundles.size();
+            expectedOps = nodeBundles.size() + remoteClusterIndices.size();
 
             nodesResponses = new AtomicReferenceArray<>(expectedOps);
         }
 
         public void start() {
-            if (nodeBundles.size() == 0) {
+            if (expectedOps == 0) {
                 // no shards
                 try {
-                    listener.onResponse(newResponse(request, new AtomicReferenceArray<>(0), true, nodeBundles));
+                    listener.onResponse(mergeResponses(request, new AtomicReferenceArray<>(0), true, nodeBundles));
                 } catch (Exception e) {
                     listener.onFailure(e);
                 }
                 // TODO or remove above try and instead just call finishHim() here? Helps keep return logic consistent
                 return;
             }
             // count the local operations, and perform the non local ones
-            int nodeIndex = -1;
+            int nodeIndex = 0;
             for (final String nodeId : nodeBundles.keySet()) {
                 if (checkForEarlyFinish()) {
                     return;
                 }
-                nodeIndex++;
                 Set<ShardId> shardIds = nodeBundles.get(nodeId);
                 if (shardIds.size() > 0) {
                     performOperation(nodeId, shardIds, nodeIndex);
                 } else {
                     // really, no shards active in this group
-                    onNoOperation(nodeId);
+                    onNodeFailure(nodeId, nodeIndex, null);
                 }
+                ++ nodeIndex;
+            }
+            // handle remote clusters
+            for (String clusterAlias : remoteClusterIndices.keySet()) {
+                performRemoteClusterOperation(clusterAlias, remoteClusterIndices.get(clusterAlias), nodeIndex);
+                ++ nodeIndex;
```

Comment:
Maybe rename `nodeIndex` now that it counts nodes and remote clusters? It may be verbose to say `nodeAndClusterIndex` so maybe something neutral like gathererIndex? Same naming question for nodeResponses.

Reply:
Addressed in https://github.com/elastic/elasticsearch/pull/73478/commits/1bbf140044547fa41b5c80a13ad2ea414753593e